tests/test_permissionable_component.cairo

```diff
@@ -0,0 +1,200 @@
+}
+
+#[test]
+#[should_panic(expected: ('Account: invalid length',))]
+fn test_when_permissioned_addresses_and_permissions_not_equal() {
+    let (contract_address, _) = __setup__();
+    let acct_dispatcher = IAccountDispatcher { contract_address: contract_address };
+
+    let owner = acct_dispatcher.owner();
+
+    let mut permission_addresses = ArrayTrait::new();
+    permission_addresses.append(ACCOUNT2.try_into().unwrap());
+    permission_addresses.append(ACCOUNT3.try_into().unwrap());
+    permission_addresses.append(ACCOUNT4.try_into().unwrap());
+
+    let mut permissions = ArrayTrait::new();
+    permissions.append(true);
+    permissions.append(true);
+
+    let permissionable_dispatcher = IPermissionableDispatcher { contract_address };
+
+    start_cheat_caller_address(contract_address, owner);
+    permissionable_dispatcher.set_permission(permission_addresses, permissions)
+}
+
+
+#[test]
+fn test_permissionable() {
+    let (contract_address, _) = __setup__();
+    let acct_dispatcher = IAccountDispatcher { contract_address: contract_address };
+
+    let owner = acct_dispatcher.owner();
+
+    let mut permission_addresses = ArrayTrait::new();
+    permission_addresses.append(ACCOUNT2.try_into().unwrap());
+    permission_addresses.append(ACCOUNT3.try_into().unwrap());
+    permission_addresses.append(ACCOUNT4.try_into().unwrap());
+
+    let mut permissions = ArrayTrait::new();
+    permissions.append(true);
+    permissions.append(true);
+    permissions.append(true);
+
+    start_cheat_caller_address(contract_address, owner);
+
+    let permissionable_dispatcher = IPermissionableDispatcher { contract_address };
+    permissionable_dispatcher.set_permission(permission_addresses, permissions);
+
+    let has_permission = permissionable_dispatcher
+        .has_permission(owner, ACCOUNT2.try_into().unwrap());
+
+    assert(has_permission == true, 'Account: not permitted');
+    stop_cheat_caller_address(contract_address);
+}
+
+#[test]
+fn test_not_permitted() {
+    let (contract_address, _) = __setup__();
+    let acct_dispatcher = IAccountDispatcher { contract_address: contract_address };
+
+    let owner = acct_dispatcher.owner();
+
+    let mut permission_addresses = ArrayTrait::new();
+    permission_addresses.append(ACCOUNT2.try_into().unwrap());
+    permission_addresses.append(ACCOUNT3.try_into().unwrap());
+    permission_addresses.append(ACCOUNT4.try_into().unwrap());
+
+    let mut permissions = ArrayTrait::new();
+    permissions.append(true);
+    permissions.append(true);
+    permissions.append(false);
+
+    start_cheat_caller_address(contract_address, owner);
+
+    let permissionable_dispatcher = IPermissionableDispatcher { contract_address };
+    permissionable_dispatcher.set_permission(permission_addresses, permissions);
+
+    let has_permission = permissionable_dispatcher
+        .has_permission(owner, ACCOUNT4.try_into().unwrap());
+
+    assert(has_permission == false, 'Account: permitted');
```

Comment:
add checks for all 3 accounts in the array, not just ACCOUNT4

Reply:
ok